src/plugins/muc-views/modals/templates/occupant.js

```diff
@@ -2,10 +2,168 @@ import 'shared/avatar/avatar.js';
+    const ownAffiliation = muc.getOwnAffiliation();
+
+    let handleBlock = (ev) => {
+        api.blockUser([jid]);
+    };
+    let handleUnblock = (ev) => {
+        api.unblockUser([jid]);
+    };
+    let handleKick = (ev) => {
+        setRole(muc, 'kick', jid, [], ['moderator']);
+    };
+    let handleMute = (ev) => {
+        setRole(muc, 'mute', jid, [], ['moderator']);
+    };
+    let handleVoice = (ev) => {
+        setRole(muc, 'voice', jid, [], ['moderator']);
+    };
+    let handleOp = (ev) => {
+        setRole(muc, 'op', jid, ['admin', 'owner'], ['moderator']);
+    };
+    let handleDeOp = (ev) => {
+        setRole(muc, 'deop', jid, ['admin', 'owner'], ['moderator']);
+    };
+    let handleBan = (ev) => {
+        verifyAndSetAffiliation(muc, 'ban', jid, ["admin", "owner"]);
+    };
+    let handleMember = (ev) => {
+        verifyAndSetAffiliation(muc, 'member', jid, ["admin", "owner"]);
+    };
+    let handleAdmin = (ev) => {
+        verifyAndSetAffiliation(muc, 'admin', jid, ["admin", "owner"]);
+    };
+    let handleOwner = async (ev) => {
+        const confirmed = await _converse.api.confirm("Are you sure you want to promote?",
+                ["Promoting a user to owner may be irreversible.",
+                 "Only server administrators may demote an owner of a Multi User Chat."],
+                []).then((x) => x.length === 0);
+        if (confirmed) {
+                verifyAndSetAffiliation(muc, 'owner', jid, ["admin", "owner"]);
+        } else {
+                showOccupantModal(ev, model);
+        }
+    };
+
+    const blockButton = html`<button class='btn btn-primary' @click=${handleBlock}>Block</button>`
+    const unblockButton = html`<button class='btn btn-primary' @click=${handleUnblock}>Unblock</button>`
+    const banButton = html`<button class='btn btn-primary' @click=${handleBan}>Ban</button>`
+    const kickButton = html`<button class='btn btn-primary' @click=${handleKick}>Kick</button>`
+
+    const muteButton = html`<button class='btn btn-primary' @click=${handleMute}>Mute</button>`
+    const unmuteButton = html`<button class='btn btn-primary' @click=${handleVoice}>Unmute</button>`
+    const memberButton = (memberText) => html`<button class='btn btn-primary' @click=${handleMember}>${memberText}</button>`
+    const addToChatButton   = memberButton("Add to Chat");
+    const unbanButton       = memberButton("Unban");
+    const removeAdminButton = memberButton("Remove Admin Status");
+
+    const opButton = html`<button class='btn btn-primary' @click=${handleOp}>Make Moderator</button>`
+    const deOpButton = html`<button class='btn btn-primary' @click=${handleDeOp}>Remove Moderator Status</button>`
+    const adminButton = html`<button class='btn btn-primary' @click=${handleAdmin}>Make Admin</button>`
+    const ownerButton = html`<button class='btn btn-primary' @click=${handleOwner}>Make Owner</button>`
+
+    // The following table stores a map from Affiliation x Role -> Button
```

Comment:
Should clarify, the mapping is from OwnAffiliation x { Target's Current Affiliation, Target's Current Role } -> Button